api/src/main/resources/Appointment.hbm.xml

```diff
@@ -50,6 +50,7 @@
         <many-to-one name="creator" class="org.openmrs.User" column="creator" not-null="true"/>
         <many-to-one name="relatedAppointment" class="org.openmrs.module.appointments.model.Appointment"
                     column="related_appointment_id" unique="true"/>
+        <property name="dateHonored" type="java.util.Date" column="date_honored"/>
```

Comment:
I'm a little hesitant about this. It seems like it would be better to have a link from a Visit -> an Appointment (which could be done via a Visit attribute) rather than tracking state that changes the lifetime of the appointment. This is less likely to lead to inconsistent data (where, for example, the appointments table indicates _X_ # of appointments were fulfilled, but there are _Y_ visits recorded in the system).

Reply:
Thanks Ian. Our intention was to have a sure way of letting us know those who come early for their appointments. We probably can keep track of visits and mark appointments as fulfilled if a patient visits just before the appointment date but again that is never 100%. We added a section in the check-in form that lists all future appointments for a patient with a checkbox to mark if any of the future appointments is being fulfilled by the visit. This way, it is easy to report those who come early without necessarily doing a lot of guesses. How else would you approach this?